tests/tests_geomstats/test_spd_matrices.py

```diff
@@ -19,559 +21,548 @@
+                    n=2,
+                    power_euclidean=1.0,
+                    tangent_vec=[[2.0, 0.0], [0.0, 2.0]],
+                    base_point=[[1.0, 0.0], [0.0, 1.0]],
+                    expected=[[3.0, 0.0], [0.0, 3.0]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def log_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    power_euclidean=1.0,
+                    point=[[2.0, 0.0], [0.0, 2.0]],
+                    base_point=[[1.0, 0.0], [0.0, 1.0]],
+                    expected=[[1.0, 0.0], [0.0, 1.0]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def log_exp_composition_data(self):
+            power_euclidean = [1.0] + [random.uniform(-1, 1) for l in range(25)]
+            return self.log_exp_composition(
+                SPDMatrices, power_euclidean=power_euclidean
+            )
+
+    testing_data = TestDataSPDMetricEuclidean()
+
+    def test_inner_product(
+        self, n, power_euclidean, tangent_vec_a, tangent_vec_b, base_point, expected
+    ):
+        metric = SPDMetricEuclidean(n, power_euclidean)
+        result = metric.inner_product(
+            gs.array(tangent_vec_a), gs.array(tangent_vec_b), gs.array(base_point)
         )
-        metric = self.metric_affine
-
-        # Test with the 1 base_point, and several different tangent_vecs
-        result = metric.exp(n_tangent_vec_same_base, one_base_point)
-
-        self.assertAllClose(gs.shape(result), (n_samples, self.space.n, self.space.n))
-
-        # Test with the same number of base_points and tangent_vecs
-        result = metric.exp(n_tangent_vec, n_base_point)
+        self.assertAllClose(result, gs.array(expected))
 
-        self.assertAllClose(gs.shape(result), (n_samples, self.space.n, self.space.n))
-
-    def test_log_vectorization(self):
-        """Test of SPDMetricAffine.log with power 1 and vectorization."""
-        n_samples = self.n_samples
-        one_base_point = self.space.random_point(n_samples=1)
-        n_base_point = self.space.random_point(n_samples=n_samples)
-
-        one_point = self.space.random_point(n_samples=1)
-        n_point = self.space.random_point(n_samples=n_samples)
-        metric = self.metric_affine
-
-        # Test with different points, one base point
-        result = metric.log(n_point, one_base_point)
-
-        self.assertAllClose(gs.shape(result), (n_samples, self.space.n, self.space.n))
-
-        # Test with the same number of points and base points
-        result = metric.log(n_point, n_base_point)
-
-        self.assertAllClose(gs.shape(result), (n_samples, self.space.n, self.space.n))
-
-        # Test with the one point and n base points
-        result = metric.log(one_point, n_base_point)
-
-        self.assertAllClose(gs.shape(result), (n_samples, self.space.n, self.space.n))
-
-    def test_geodesic_and_belongs(self):
```

Comment:
Can we test the geodesic function? Within the randomized tests, with the belongs function, if that is easier.